### ISSUE TITLE
Dead code in maybeDeferred()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3894]
|<img alt="amaury's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| amaury reported|
|-|-|
|Trac ID|trac#3894|
|Type|enhancement|
|Created|2009-06-29 11:41:54Z|

In `defer.maybeDeferred`, the variable `deferred` is no more useful: it is always set to `None`, and the `else:` clause of the exception returns in every path.  Patch attached.

Attachments:

* [maybeDeferred.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a4e/a4ea66607c65d411e1e60e6673e1a49c9677c827/2d4b13b218471899804ed56cd341f7cf804440b7.patch) (882 bytes) - added by amaury on 2009-06-29 11:42:09Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3894 3894
type__enhancement enhancement
reporter__amaury amaury
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1246275714000000 1246275714000000
changetime__1246313574000000 1246313574000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_2"></a>|@mwhudson commented|
|-|-|

Yes, that all makes sense.  I'll commit it.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_3"></a>|@mwhudson set status to `closed`|
|-|-|

(In [27030]) Apply maybeDeferred.patch

Author: amaury
Reviewer: mwh
Fixes: [#3894](https://github.com/twisted/twisted/issues/3894)

Remove some dead code in maybeDeferred.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
